web/docs/auth/email.md

```diff
@@ -595,7 +584,234 @@ Users can enter their new password there.
 
 The content of the e-mail can be customized, read more about it [here](#passwordreset-passwordresetconfig-).
 
-## Using The Auth
+## Creating a Custom Sign-up Action
+
+> TODO: test out the code below and make sure it works
+
+:::caution Creating a custom sign-up action
+
+We don't recommend creating a custom sign-up action unless you have a good reason to do so. It is a complex process and you can easily make a mistake that will compromise the security of your app.
+:::
+
+The code of your custom sign-up action can look like this:
+
+<Tabs groupId="js-ts">
+<TabItem value="js" label="JavaScript">
+
+```wasp title="main.wasp"
+// ...
+
+action customSignup {
+  fn: import { signup } from "@server/auth/signup.js",
+  entities: [User]
+}
+```
+
+
+```js title="src/server/auth/signup.js"
+import {
+  ensurePasswordIsPresent,
+  ensureValidPassword,
+  ensureValidEmail,
+} from '@wasp/auth/validation.js'
+import {
+  createProviderId,
+  sanitizeAndSerializeProviderData,
+  deserializeAndSanitizeProviderData,
+  findAuthIdentity,
+  createUser,
+} from '@wasp/auth/utils.js'
+import {
+  createEmailVerificationLink,
+  sendEmailVerificationEmail,
+} from '@wasp/auth/providers/email/utils.js'
+
+export const signup = async (args, { entities: { User } }) => {
```

Comment:
I'm not sure, maybe I mistyped the title and put "login" in it. 

We should probably wait for users to need the custom login action IMHO since I haven't come up with a use-case myself.

Reply:
I've removed the "Login" mention.